src/schema/notifications.ts

```diff
@@ -422,6 +423,22 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         });
       }
 
+      const newCommentComments = data.filter(({ notificationType }) =>
+        commentReplyNotificationTypes.includes(notificationType),
+      );
+      if (newCommentComments.length) {
+        const commentIds = newCommentComments.map(
+          ({ referenceId }) => referenceId,
+        );
+        const commentComments = await ctx
+          .getRepository(Comment)
+          .find({ select: ['id', 'parentId'], where: { id: In(commentIds) } });
+        commentComments.forEach(({ id, parentId }) => {
+          const param = params.find(({ referenceId }) => referenceId === id);
+          param!.referenceId = parentId || id;
```

Comment:
```suggestion
          if (!param) {
              return
          }

          param.referenceId = parentId || id;
```

Reply:
This doesn't work, we need to always set it regardless of it existing or not.
In your version we basically never set the parent.

Reply:
But you should not need to force coercion with `!.` If `param` is `undefined` the code will break.